packages/kbn-handlebars/src/visitor.ts

```diff
@@ -156,13 +139,16 @@ export class ElasticHandlebarsVisitor extends Handlebars.Visitor {
   render(context: any, options: ExtendedRuntimeOptions = {}): string {
     this.contexts = [context];
     this.output = [];
-    this.runtimeOptions = Object.assign({}, options);
-    this.container.helpers = Object.assign(this.initialHelpers, options.helpers);
-    this.container.partials = Object.assign(this.initialPartials, options.partials);
-    this.container.decorators = Object.assign(
-      this.initialDecorators,
-      options.decorators as DecoratorsHash
-    );
+    this.runtimeOptions = { ...options };
+    this.container.helpers = {
+      ...this.env.helpers,
+      ...(options.helpers as HelpersHash),
```

Comment:
__question:__ I'm wondering if we should just change `HelpersHash` interface to be compatible with `RuntimeOptions.helpers` type and avoid force-casting here (and it's not entirely correct since `options.helpers` can be `undefined`) unless there is a value in keeping `HelperDelegate`??
```ts
export interface HelpersHash {
  [name: string]: Function;
}
```

Reply:
I've refactored the types quite significantly in the upcoming chained PR #150520, which will remove the need to force cast helpers